plugins/extractors/date/date.go

```diff
@@ -0,0 +1,81 @@
+package date
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"github.com/odpf/meteor/core/extractor"
+	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/proto/odpf/meta/facets"
+	"time"
+)
+
+const (
+	MaxDates = 3
+)
+
+type Extractor struct {
+	logger plugins.Logger
+}
+
+type Payload struct{
+	Value string
+}
+
+type Payloads []Payload
+
+func (p *Payload) ToJSON() ([]byte, error) {
+	return json.Marshal(p)
+}
+
+func (p *Payload) ToProto() ([]byte, error) {
```

Comment:
For any user to writing a metadata extractor, we want them to extract in a proper schema. e.g. dataset or a user. we want this data to be mapped to a model. Wire format and sink format could be anything JSON, text, proto, or anything but ideally, extractor schema should be consistent with one of the models. So with that in mind, does it make sense to stick that users use one of the proto-model to model their metadata and codec can be a central module which allows sinks to process and dump in whatever format they would like to choose?

Reply:
Ok so I would propose that serialization should not be the concern of extractors and processors, it should be the concern for sinks. We could have a serializer module that would do the heavy lifting for serialization, so even an individual sink should not care about it and just the module.

What to serialise data into should not be decided by the extractor, it should be up to the user. For example it would be totally possible to use kafka sink and serialise it into json format, or maybe sinking it to stdout and serialize it to protobuf (not sure why users would do that but again we don't care).

Since all the serialization logic would be on the serializer module, even if we somehow have to do 10-20 switch cases, it is still being contained there and other sinks can just use it without having to rewrite the same thing.

Building proto key can also be handled in that same module, and since these metadata models (Table, Dashboard, etc) are maintained by us, it would make sense as well that we decide what would be the proto key. I dont see a reason to make it customisable for now.

Sample serializer usage in recipe
```
source:
  type: bigquery
  config: {}
sinks:
 - name: kafka
    serializer: json
    config:
      brokers: localhost:9092,localhost:9093
      topic: bigquerytable_metadata
 - name: console
    serializer: protobuf
```

what do you guys think? @ravisuhag @kushsharma